app/components/impagination-dataset.js

```diff
@@ -1 +1,116 @@
-export { default } from 'ember-impagination/components/impagination-dataset';
+import Ember from 'ember';
+import Dataset from 'npm:impagination';
+import layout from '../templates/components/impagination-dataset';
+
+const { get, set } = Ember;
+const { String: { dasherize } } = Ember;
+
+
+// TODO: Since `records` are wrapped in Ember.A() not all these methods are accurate
+// We should include Ember.Array methods here instead
+const accessors = ['concat', 'includes', 'join', 'slice', 'toString', 'toLocaleString', 'indexOf', 'lastIndexOf'];
+
+const iterators = ['forEach', 'every', 'some', 'filter', 'find', 'findIndex', 'keys', 'map', 'reduce', 'reduceRight', 'values'];
+
+const enumerables = ['objectAt'];
+
+let impaginationGetters = [
+  // Coming Soon in impagination@1.0.0-alpha.3
+  // 'hasUnrequested', 'hasRequested', 'hasPending', 'hasResolved', 'hasRejected', 'hasUnfetchable',
+  'unrequested', 'requested', 'pending', 'resolved', 'rejected', 'unfetchable',
+  'pages', 'length'
+];
+
+export default Ember.Component.extend({
+  layout: layout,
+  'load-horizon': null,
+  'unload-horizon': Infinity,
+  'page-size': null,
+  'fetch': null,
+  'filter': null,
+  'on-init': Ember.K,
```

Comment:
I think that `Ember.K` is getting deprecated, but luckily, there is an equally concise syntax you can use:

```js
'on-init'() {},
'on-observe'() {}
```